web3/index.js

```diff
@@ -35,24 +35,87 @@ isIntegrationEnabled()
         loadScripts(config);
       });
 
+      resizeParityBar();
       handleResizeEvents();
     }
   });
 
+function getParityBarElement () {
+  return document.querySelector('#container > div > div');
+}
+
 /**
  *  Propagates opening events to upper frame
  */
 function handleResizeEvents () {
   document.body.addEventListener('parity.bar.visibility', (ev) => {
-    document.querySelector('#container > div > div').style.maxHeight = '100vh';
-    window.parent.postMessage({
+    const { opened } = ev.detail;
+    const parityBarElement = getParityBarElement();
+    const message = {
       type: EV_SIGNER_BAR,
-      opened: ev.detail.opened
-    }, '*');
+      opened
+    };
+
+    parityBarElement.style.maxHeight = '100vh';
+
+    // Resize the iframe if it's closing
+    if (!opened) {
+      return window.setTimeout(() => {
+        resizeParityBar();
+      }, 100);
+    }
+
+    window.parent.postMessage(message, '*');
   });
 }
 
 /**
+ * Resize and position the iframe according
+ * to the Parity Bar style
+ */
+function resizeParityBar (wait = 2000) {
+  const parityBarElement = getParityBarElement();
+
+  // Try again in 100ms
+  if (!parityBarElement && wait > 0) {
```

Comment:
I added a longer comment. It is because on load, the parity bar is not available immediately. It takes time to show it, so we try every 100ms for 2 seconds to see if it's loaded.